sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/unit/tests/HadoopInputFormatIOTest.java

```diff
@@ -0,0 +1,796 @@
+      assertEquals(HIFSource, source);
+    }
+  }
+
+
+  /**
+   * This test validates behavior of {@link HadoopInputFormatBoundedSource#createReader()
+   * createReader()} method when {@link HadoopInputFormatBoundedSource#splitIntoBundles()
+   * splitIntoBundles()} is not called.
+   */
+  @Test
+  public void testCreateReaderIfSplitIntoBundlesNotCalled() throws Exception {
+    HadoopInputFormatBoundedSource<String, String> parentHIFSource =
+        new HadoopInputFormatBoundedSource<String, String>(
+            serConf,
+            StringUtf8Coder.of(),
+            StringUtf8Coder.of(),
+            myKeyTranslate,
+            myValueTranslate,
+            null); // Initially split is null as source is not split.
+    thrown.expect(IOException.class);
+    thrown.expectMessage(HadoopInputFormatIOContants.CREATEREADER_UNSPLIT_SOURCE_ERROR_MSG);
+    parentHIFSource.createReader(p.getOptions());
+  }
+
+  /**
+   * This test validates behavior of {@link HadoopInputFormatBoundedSource#getEstimatedSizeBytes()
+   * getEstimatedSizeBytes()} and {@link HadoopInputFormatBoundedSource#splitIntoBundles()
+   * splitIntoBundles()} when Hadoop InputFormat's {@link InputFormat#getSplits() getSplits()}
+   * returns empty list.
+   */
+  @Test
+  public void testSplitIntoBundlesIfGetSplitsReturnsEmptyList() throws Exception {
+    HadoopInputFormatBoundedSource<Text, Employee> parentHIFSource = getTestHIFSource(
+            BadEmptySplitsInputFormat.class,
+            Text.class,
+            Employee.class,
+            WritableCoder.of(Text.class),
+            AvroCoder.of(Employee.class));
+    thrown.expect(IOException.class);
+    thrown.expectMessage(HadoopInputFormatIOContants.COMPUTESPLITS_EMPTY_SPLITS_ERROR_MSG);
+    parentHIFSource.getEstimatedSizeBytes(p.getOptions());
+    thrown.expect(IOException.class);
+    thrown.expectMessage(HadoopInputFormatIOContants.COMPUTESPLITS_EMPTY_SPLITS_ERROR_MSG);
+    parentHIFSource.splitIntoBundles(0, p.getOptions());
+  }
+
+  /**
+   * This test validates behavior of {@link HadoopInputFormatBoundedSource#getEstimatedSizeBytes()
+   * getEstimatedSizeBytes()} and {@link HadoopInputFormatBoundedSource#splitIntoBundles()
+   * splitIntoBundles()} when Hadoop InputFormat's {@link InputFormat#getSplits() getSplits()}
+   * returns NULL value.
+   */
+  @Test
+  public void testSplitIntoBundlesIfGetSplitsReturnsNullValue() throws Exception {
+    HadoopInputFormatBoundedSource<Text, Employee> parentHIFSource = getTestHIFSource(
+            BadNullSplitsInputFormat.class,
+            Text.class,
+            Employee.class,
+            WritableCoder.of(Text.class),
+            AvroCoder.of(Employee.class));
+    thrown.expect(IOException.class);
+    thrown.expectMessage(HadoopInputFormatIOContants.COMPUTESPLITS_NULL_GETSPLITS_ERROR_MSG);
+    parentHIFSource.getEstimatedSizeBytes(p.getOptions());
+
+    thrown.expect(IOException.class);
```

Comment:
We added a separate tests for computeSplits() where the exception was thrown - testComputeSplitsIfGetSplitsReturnsEmptyList(),  testComputeSplitsIfGetSplitsReturnsNullValue(), testComputeSplitsIfGetSplitsReturnsListHavingNullValues()